client/gutenberg/extensions/mailchimp/edit.jsx

```diff
@@ -182,36 +184,43 @@ class MailchimpSubscribeEdit extends Component {
 				</PanelBody>
 			</InspectorControls>
 		);
+		const emailInputId = `${ classPrefix }email_input-${ instanceId }`;
 		const blockContent = (
 			<div className={ className }>
 				{ ! audition && (
-					<form ref={ this.formRef }>
-						<TextControl
-							disabled
-							placeholder={ emailPlaceholder }
-							onChange={ () => false }
-							title={ __( 'You can edit the email placeholder in the sidebar.' ) }
-							type="email"
-						/>
-						<div className="wp-block-jetpack-mailchimp_button">
-							<RichText
-								formattingControls={ [] }
-								placeholder={ __( 'Add button text…' ) }
-								value={ submitLabel }
-								onChange={ value => setAttributes( { submitLabel: value } ) }
+					<Fragment>
+						<p>
```

Comment:
Nitpick -- do we need the wrapping `<p />`s? Feels a bit like hitting Enter in a Word doc to adjust vertical spacing :sweat_smile:

Reply:
We're wrapping the elements in `<p/>`s on the front end to combat a whole series of visual problems. I did them in the editor too just to roughly mirror the markup. Are you suggesting we shouldn't be doing this in either setting, or just in editor?

Reply:
I was suggesting doing it in neither setting -- but if they're an easy fix for an otherwise complicated issue, I guess it's fine to keep them.

Reply:
This was @thomasguillot's solution to a whole series of issues that were raised in the CfT. We opted to minimize the frontend styling of the block's form elements, leaving this up to the theme. However, this led to some minor visual problems like 0 spacing between input and button, cut-off placeholder text, etc. Wrapping them in `<p/>` tags provided a nice easy way to protect against all of that without getting too opinionated about styling.

Reply:
Looks like we cannot use `p` because some components render `div` elements and `<p><div/></p>` (div not being an inline element) isn't allowed in HTML.

https://github.com/Automattic/wp-calypso/pull/30970#issuecomment-467328194

I'll remove `p`s and we can follow up with some CSS to fix those visual issues in the editor.

Reply:
Fixed in https://github.com/Automattic/wp-calypso/pull/30970/commits/ac0660a7a27debace40aed6e97d422c0d2aab12c

This is how block looks like in Storefront theme currently:

<img width="517" alt="image" src="https://user-images.githubusercontent.com/87168/53395938-d5c29200-39ab-11e9-8311-af8c29c0682d.png">

Reply:
Looks like this caused regression (via https://github.com/Automattic/jetpack/pull/11403#pullrequestreview-207862434)

> In the past when I would customize the processing text, the whole form would disappear. Now it does not:
>
> ![image](https://user-images.githubusercontent.com/426388/53404343-c34d5600-39b5-11e9-999d-da7d9461daed.png)
> ![image](https://user-images.githubusercontent.com/426388/53404374-d52ef900-39b5-11e9-878c-7ea8337457b3.png)

...because hiding is done by hiding `p` — https://github.com/Automattic/wp-calypso/pull/30970/files#diff-8bd401a2fae5a72e7ca5e11c76e89b65R16

```
&.wp-block-jetpack-mailchimp_notication-audition p {
	display: none;
}
```

Feels like we can have a little bit more robust way to do this. :-)

Reply:
Regression fixed in https://github.com/Automattic/wp-calypso/pull/30970/commits/21818816cc63ea5166e2d64ed120322a83f9e83a